framework/components/dockercompose/chip_ingress_set/chip_ingress.go

```diff
@@ -0,0 +1,261 @@
+	ExtraDockerNetworks []string `toml:"extra_docker_networks"`
+	Output              *Output  `toml:"output"`
+	UseCache            bool     `toml:"use_cache"`
+}
+
+func defaultChipIngress(in *Input) *Input {
+	if in.ComposeFile == "" {
+		in.ComposeFile = "./docker-compose.yml"
+	}
+	return in
+}
+
+const (
+	DEFAULT_STACK_NAME = "chip-ingress"
+
+	DEFAULT_CHIP_INGRESS_GRPC_PORT    = "50051"
+	DEFAULT_CHIP_INGRESS_SERVICE_NAME = "chip-ingress"
+
+	DEFAULT_RED_PANDA_SCHEMA_REGISTRY_PORT = "18081"
+	DEFAULT_RED_PANDA_KAFKA_PORT           = "19092"
+	DEFAULT_RED_PANDA_SERVICE_NAME         = "redpanda-0"
+
+	DEFAULT_RED_PANDA_CONSOLE_SERVICE_NAME = "redpanda-console"
+	DEFAULT_RED_PANDA_CONSOLE_PORT         = "8080"
+)
+
+func New(in *Input) (*Output, error) {
+	if in == nil {
+		return nil, errors.New("input is nil")
+	}
+
+	if in.UseCache {
+		if in.Output != nil {
+			return in.Output, nil
+		}
+	}
+
+	in = defaultChipIngress(in)
+	identifier := framework.DefaultTCName(DEFAULT_STACK_NAME)
+	framework.L.Debug().Str("Compose file", in.ComposeFile).Msgf("Starting Chip Ingress stack with identifier %s", framework.DefaultTCName(DEFAULT_STACK_NAME))
+
+	composeFilePath, fileErr := composeFilePath(in.ComposeFile)
+	if fileErr != nil {
+		return nil, errors.Wrap(fileErr, "failed to get compose file path")
+	}
+
+	stack, stackErr := compose.NewDockerComposeWith(
+		compose.WithStackFiles(composeFilePath),
+		compose.StackIdentifier(identifier),
+	)
+	if stackErr != nil {
+		return nil, errors.Wrap(stackErr, "failed to create compose stack for Chip Ingress")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	upErr := stack.
+		WithEnv(map[string]string{
+			"BASIC_AUTH_ENABLED": "false",
+			"BASIC_AUTH_PREFIX":  "",
+		}).
+		Up(ctx)
+
+	if upErr != nil {
+		return nil, errors.Wrap(upErr, "failed to start stack for Chip Ingress")
+	}
+
+	stack.WaitForService(DEFAULT_CHIP_INGRESS_SERVICE_NAME,
+		wait.ForLog("GRPC server is live").WithPollInterval(200*time.Millisecond).WithStartupTimeout(1*time.Minute),
+	)
+
+	chipIngressContainer, ingressErr := stack.ServiceContainer(ctx, DEFAULT_CHIP_INGRESS_SERVICE_NAME)
+	if ingressErr != nil {
+		return nil, errors.Wrap(ingressErr, "failed to get chip-ingress container")
+	}
+
+	cli, cliErr := client.NewClientWithOpts(
+		client.FromEnv,
+		client.WithAPIVersionNegotiation(),
+	)
+	if cliErr != nil {
+		return nil, errors.Wrap(cliErr, "failed to create docker client")
+	}
+	defer cli.Close()
+
+	timeout := time.After(1 * time.Minute)
+	tick := time.Tick(500 * time.Millisecond)
+
+	// so let's try to connect to a Docker network a couple of times, there must be a race condition in Docker
+	// and even when network sandbox has been created and container is running, this call can still fail
+	// retrying is simpler than trying to figure out how to correctly wait for the network sandbox to be ready
+	var connectNetwork = func(networkName string) error {
```

Comment:
Can't we just be on the same network? Seems like we either need to remove the default network from CTF or to configure this compose to use the same CTF network

Reply:
and what's the problem with this approach?